src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/AccountSupervisor.kt

```diff
@@ -767,20 +769,29 @@ internal open class AccountSupervisor(
             val result = helper.parser.decodeJsonObject(additionalPayload)
 
             if (error == null && result != null) {
-                val url = complianceGeoblockUrl()
                 val signedMessage = helper.parser.asString(result["signedMessage"])
                 val publicKey = helper.parser.asString(result["publicKey"])
                 val timestamp = helper.parser.asString(result["timestamp"])
+                val isKeplr = helper.parser.asBool(result["isKeplr"])
+                val url = if (isKeplr == true) complianceGeoblockKeplrUrl() else complianceGeoblockUrl()
 
                 val isUrlAndKeysPresent =
                     url != null &&
                         signedMessage != null &&
-                        publicKey != null &&
-                        timestamp != null
+                        publicKey != null && (timestamp != null || isKeplr == true)
+
                 val isStatusValid = status != ComplianceStatus.UNKNOWN
 
-                if (isUrlAndKeysPresent && isStatusValid) {
-                    val body: IMap<String, String> =
+                val body: Map<String, String> =
+                    if (isKeplr == true) {
+                        iMapOf(
+                            "address" to address.rawAddress,
+                            "message" to message,
+                            "action" to complianceAction.toString(),
+                            "signedMessage" to signedMessage!!,
```

Comment:
[q] why do these properties have to be assumed non-null? (i know it's part of previously committed code too, but curious if you have the answer)

Reply:
Just that the request needs all these fields.

I check for it to be not null with `isUrlAndKeysPresent` before the body gets used, so if anything is null, the request would not send.

I don't know if there is a better kotlin way to do this tho

Reply:
oh, does kotlin remove the field entirely if the value is null?

Reply:
oh it will actually throw, thats a good catch actually. I think it should be inside the if statement